packages/salesforcedx-vscode-replay-debugger/src/adapter/apexReplayDebug.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { EOL } from 'os';
+import {
+  DebugSession,
+  OutputEvent,
+  Source,
+  StoppedEvent,
+  Thread,
+  ThreadEvent
+} from 'vscode-debugadapter';
+import { DebugProtocol } from 'vscode-debugprotocol';
+import { LogFile } from '../core/logFile';
+import { nls } from '../messages';
+
+export interface LaunchRequestArguments
+  extends DebugProtocol.LaunchRequestArguments {
+  logFile: string;
+  stopOnEntry: boolean | true;
+  traceLog: boolean | true;
+}
+
+export class ApexReplayDebug extends DebugSession {
+  public static THREAD_ID = 1;
+  protected logFile: LogFile;
+
+  constructor() {
+    super();
+    this.setDebuggerLinesStartAt1(true);
+    this.setDebuggerPathFormat('uri');
+  }
+
+  public initializeRequest(
+    response: DebugProtocol.InitializeResponse,
+    args: DebugProtocol.InitializeRequestArguments
+  ): void {
+    response.body = {
+      supportsCompletionsRequest: false,
+      supportsConditionalBreakpoints: false,
+      supportsDelayedStackTraceLoading: false,
+      supportsEvaluateForHovers: false,
+      supportsExceptionInfoRequest: false,
+      supportsExceptionOptions: false,
+      supportsFunctionBreakpoints: false,
+      supportsHitConditionalBreakpoints: false,
+      supportsLoadedSourcesRequest: false,
+      supportsRestartFrame: false,
+      supportsSetVariable: false,
+      supportsStepBack: false,
+      supportsStepInTargetsRequest: false
+    };
+    response.success = true;
+    this.sendResponse(response);
+  }
+
+  public launchRequest(
+    response: DebugProtocol.LaunchResponse,
+    args: LaunchRequestArguments
+  ): void {
+    this.logFile = new LogFile(args);
+    if (!this.logFile.hasLogLines()) {
+      response.success = false;
+      response.message = nls.localize('no_log_file_text');
+      this.sendResponse(response);
+      return;
+    }
+    if (args.stopOnEntry) {
```

Comment:
Here's the part where it decides to stop on the first line of debug log or continue until the first breakpoint.